cmd/api/src/services/fileupload/validation.go

```diff
@@ -14,49 +14,50 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-package datapipe
+package fileupload
 
 import (
 	"encoding/json"
 	"errors"
-	"fmt"
 	"github.com/specterops/bloodhound/log"
+	"github.com/specterops/bloodhound/src/model/ingest"
 	"io"
 )
 
-func ValidateMetaTag(reader io.ReadSeeker) (Metadata, error) {
-	_, err := reader.Seek(0, io.SeekStart)
-	if err != nil {
-		return Metadata{}, fmt.Errorf("error seeking to start of file: %w", err)
-	}
+var ZipMagicBytes = []byte{0x50, 0x4b, 0x03, 0x04}
+
+// ValidateMetaTag ensures that the correct tags are present in a json file for data ingest.
+// If readToEnd is set to true, the stream will read to the end of the file (needed for TeeReader)
+func ValidateMetaTag(reader io.Reader, readToEnd bool) (ingest.Metadata, error) {
 	var (
 		depth            = 0
 		decoder          = json.NewDecoder(reader)
 		dataTagFound     = false
 		dataTagValidated = false
 		metaTagFound     = false
-		meta             Metadata
+		meta             ingest.Metadata
 	)
 
 	for {
 		if dataTagValidated && metaTagFound {
-			return meta, nil
+			break
 		}
```

Comment:
This pattern took me a second to parse that the success kicks out in the subsequent loop iteration. What do you think of moving this to the end of the for loop for readability?